### ISSUE TITLE
key-spacing: `mode` and `on` incorrect behavior

### ISSUE THREAD

**Issue by p1100i** [eslint/eslint#11321]
**Env**
* eslint: v5.12.1
* node: v11.8.0
* npm: 6.7.0
* default parser

**Code**
```js
// Content of src.js
/* eslint key-spacing: [2, { align: { "mode" : "minimum", "on" : "colon" } }] */

var obj = {
  'firstProp'    : 1,
  'secondProp': 2,
  'lastProp'     : 3
}
```

**Behavior**
* **expected**
I've expected that `eslint --no-eslintrc src.js` exit code is a non-zero value (it fails).

* **current**
`eslint --no-eslintrc src.js` exit code is 0.

**Details**
* some related issues: [3120](https://github.com/eslint/eslint/issues/3120), [3363](https://github.com/eslint/eslint/issues/3363), [3453](https://github.com/eslint/eslint/issues/3453)
* also I was hoping with this config such snippet will pass (but it doesn't)
```js
    var obj = {
      'firstProp'    : 1,
      'secondProp'   : 2,
      'lastProp'     : 3
    }
```
